benchmark.py

```
@@ -0,0 +1,833 @@
+def main() -> None:
+    config = create_config_from_args()
+    
+    if not os.path.exists(config.server_exe):
+        print(f"Error: {config.server_exe} not found. Please ensure the project is built with 'zig build --release=fast'.")
+        raise SystemExit(1)
+    
+    try:
+        asyncio.run(run_benchmark(config))
+    except KeyboardInterrupt:
+        print("\nBenchmark interrupted by user.")
+        raise SystemExit(1)
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)